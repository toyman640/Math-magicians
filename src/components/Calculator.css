.Container {
  width: 25%;
  margin-left: 37.5%;
  margin-top: 10%;
}

.Screen {
  color: #fff;
  height: 50px;
  background-color: #50505050;
}

.Digit {
  align-items: center;
  float: right;
  line-height: 50px;
  font-size: 20px;
  margin-right: 1%;
}

.Row {
  margin-top: 0;
  padding-top: 0;
}

.Other {
  width: 25%;
  margin-top: 0;
  padding-top: 0;
  border: 0;
  background-color: #d4d4d2;
  height: 50px;
  border-right: 1px solid rgb(138, 137, 137);
  border-bottom: 1px solid rgb(138, 137, 137);
}

.Operator {
  width: 25%;
  margin-top: 0;
  padding-top: 0;
  border: 0;
  height: 50px;
  border-right: 1px solid rgb(138, 137, 137);
  border-bottom: 1px solid rgb(138, 137, 137);
  background: #ff9500;
}

.Zero {
  width: 50%;
  margin-top: 0;
  padding-top: 0;
  border: 0;
  background-color: #d4d4d2;
  height: 50px;
  border-right: 1px solid rgb(138, 137, 137);
  border-bottom: 1px solid rgb(138, 137, 137);
}
