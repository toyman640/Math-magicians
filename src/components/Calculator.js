function Calculator() {
  return (
    <div className="Container">
      <Screen />
      <div className="Row">
        <button className="Other" type="button">AC</button>
        <button className="Other" type="button">+/-</button>
        <button className="Other" type="button">%</button>
        <button type="button" className="Operator">&#247;</button>
      </div>
      <div className="Row">
        <button className="Other" type="button">7</button>
        <button className="Other" type="button">8</button>
        <button className="Other" type="button">9</button>
        <button type="button" className="Operator">x</button>
      </div>
      <div className="Row">
        <button className="Other" type="button">4</button>
        <button className="Other" type="button">5</button>
        <button className="Other" type="button">6</button>
        <button type="button" className="Operator">-</button>
      </div>
      <div className="Row">
        <button className="Other" type="button">1</button>
        <button className="Other" type="button">2</button>
        <button className="Other" type="button">3</button>
        <button type="button" className="Operator">+</button>
      </div>
      <div className="Row">
        <button className="Zero" type="button">0</button>
        <button className="Other" type="button">.</button>
        <button className="Operator" type="button">=</button>
      </div>
    </div>
  );
}

function Screen() {
  return (
    <div className="Screen">
      <span className="Digit">0</span>
    </div>
  );
}

export default Calculator;
