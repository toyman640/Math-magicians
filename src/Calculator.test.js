import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Calculator from './components/Calculator';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('updates the screen when buttons are clicked', () => {
  const component = renderer.create(<Calculator />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const button1 = component.root.findByProps({ children: '1' });
  button1.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const button2 = component.root.findByProps({ children: '2' });
  button2.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const button3 = component.root.findByProps({ children: '3' });
  button3.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const buttonPlus = component.root.findByProps({ children: '+' });
  buttonPlus.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const button4 = component.root.findByProps({ children: '4' });
  button4.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const buttonEquals = component.root.findByProps({ children: '=' });
  buttonEquals.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
