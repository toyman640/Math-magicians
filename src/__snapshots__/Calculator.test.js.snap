// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
Array [
  <h3
    className="Calchead"
  >
    Let's do some math!
  </h3>,
  <div
    className="Container"
  >
    <div
      className="Screen"
    >
      <span
        className="Digit"
      >
        0
      </span>
    </div>
    <div
      className="Row"
    >
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="Operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="Row"
    >
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="Operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="Row"
    >
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="Operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="Row"
    >
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="Operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="Row"
    >
      <button
        className="Zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="Other"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="Operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
